### ISSUE TITLE
Please fix OnSettingsUI

### ISSUE THREAD

**Issue by SamsamTS** [Katalyst6/CSL.TransitAddonMod#8]
Hi Katalyst6

Can you please change
`UIButton componentInChildren = Resources.FindObjectsOfTypeAll<OptionsKeymappingPanel>()[0].GetComponentInChildren<UITabstrip>().GetComponentInChildren<UIButton>();`

to 

`UIButton componentInChildren = UIView.Find<UITabstrip>("KeyMappingTabStrip").GetComponentInChildren<UIButton>();`

It is not safe. I cannot use OptionsKeymappingPanel because of this.

**Comment by Katalyst6** [comment]
UIView.Find<UITabstrip>("KeyMappingTabStrip") is not working...
If you have a better solution, i'm all hear.

Meanwhile, I might ask: why does that have an impact on anything? I mean, it's only the option panel, there are little or no reason at all to modify anything related to that...

**Comment by SamsamTS** [comment]
I copied that wrong I guess:

`UIButton componentInChildren = GameObject.Find("KeyMappingTabStrip").GetComponentInChildren<UIButton>();`

I wanted to use an instance of OptionsKeymappingPanel (actually derived from) . The problem is that your mod would find my instance in the 0 position which didn't contain any UITabStrip and would then cause a NullPointerException.

In the end I copied the entire decompiled OptionsKeymappingPanel class instead of inheriting from it, so it's all good now. I still wouldn't recommend the use of Resources.FindObjectsOfTypeAll because there is no guaranty that the first element is the one you want.

**Comment by Katalyst6** [comment]
Yeah I understand,  but we are only looking for a template here...So if the template dissappears for unknown reasons, which may still happens with the code you have provided, we will still get screwed...
Thanks for your effortCheers!<div>
</div><div>
</div><div>-------- Original message --------</div><div>From: SamsamTS notifications@github.com </div><div>Date: 2016-03-29  08:04  (GMT-05:00) </div><div>To: "Katalyst6/CSL.TransitAddonMod" CSL.TransitAddonMod@noreply.github.com </div><div>Cc: Katalyst6 david_charlebois@msn.com </div><div>Subject: Re: [Katalyst6/CSL.TransitAddonMod] Please fix OnSettingsUI (#8) </div><div>
</div>
I must have done something wrong I thought it worked. Here is something that should work:

`UIButton componentInChildren = GameObject.Find("Keymapping").GetComponentInChildren<UITabstrip>().GetComponentInChildren<UIButton>();`

I wanted to use an instance of OptionsKeymappingPanel (actually derived from) . The problem is that your mod would find my instance in the 0 position which didn't contain any UITabStrip and would then cause a NullPointerException.

In the end I copied the entire decompiled OptionsKeymappingPanel class instead of inheriting from it, so it's all good now. I still wouldn't recommend the use of Resources.FindObjectsOfTypeAll because there is no guaranty that the first element is the one you want.

---
